File: frontend/src/services/medicineService.js
import api from './api'

export const medicineService = {
  async getAll(search = '') {
    const response = await api.get('/medicines', { params: { search } })
    return response.data
  },

  async getById(id) {
    const response = await api.get(`/medicines/${id}`)
    return response.data
  },

  async create(medicine) {
    const response = await api.post('/medicines', medicine)
    return response.data
  },

  async update(id, medicine) {
    const response = await api.put(`/medicines/${id}`, medicine)
    return response.data
  },

  async delete(id) {
    const response = await api.delete(`/medicines/${id}`)
    return response.data
  }
}

File: backend/services/medicine_service.go
package services

import (
	"errors"
	"pharmacy-erp/models"
	"pharmacy-erp/repository"
)

type MedicineService struct {
	medicineRepo *repository.MedicineRepository
}

func NewMedicineService() *MedicineService {
	return &MedicineService{
		medicineRepo: repository.NewMedicineRepository(),
	}
}

func (s *MedicineService) CreateMedicine(medicine *models.Medicine) error {
	return s.medicineRepo.Create(medicine)
}

func (s *MedicineService) GetAllMedicines(search string) ([]models.Medicine, error) {
	return s.medicineRepo.FindAll(search)
}

func (s *MedicineService) GetMedicineByID(id uint) (*models.Medicine, error) {
	return s.medicineRepo.FindByID(id)
}

func (s *MedicineService) UpdateMedicine(id uint, medicine *models.Medicine) error {
	existing, err := s.medicineRepo.FindByID(id)
	if err != nil {
		return errors.New("medicine not found")
	}

	existing.Name = medicine.Name
	existing.Manufacturer = medicine.Manufacturer
	existing.BatchNumber = medicine.BatchNumber
	existing.ExpiryDate = medicine.ExpiryDate
	existing.PurchasePrice = medicine.PurchasePrice
	existing.SellingPrice = medicine.SellingPrice
	existing.Quantity = medicine.Quantity

	return s.medicineRepo.Update(existing)
}

func (s *MedicineService) DeleteMedicine(id uint) error {
	return s.medicineRepo.Delete(id)
}

func (s *MedicineService) GetLowStockMedicines(threshold int) ([]models.Medicine, error) {
	return s.medicineRepo.FindLowStock(threshold)
}
